misc/metrics/src/swarm.rs

```diff
@@ -143,13 +157,19 @@ impl<TBvEv, THandleErr> super::Recorder<libp2p_swarm::SwarmEvent<TBvEv, THandleE
     fn record(&self, event: &libp2p_swarm::SwarmEvent<TBvEv, THandleErr>) {
         match event {
             libp2p_swarm::SwarmEvent::Behaviour(_) => {}
-            libp2p_swarm::SwarmEvent::ConnectionEstablished { endpoint, .. } => {
-                self.connections_established
-                    .get_or_create(&ConnectionEstablishedLabels {
-                        role: endpoint.into(),
-                        protocols: protocol_stack::as_string(endpoint.get_remote_address()),
-                    })
-                    .inc();
+            libp2p_swarm::SwarmEvent::ConnectionEstablished {
+                endpoint,
+                time_taken,
+                ..
+            } => {
+                let labels = ConnectionEstablishedLabels {
+                    role: endpoint.into(),
+                    protocols: protocol_stack::as_string(endpoint.get_remote_address()),
+                };
+                self.connections_established.get_or_create(&labels).inc();
+                self.connection_establishment_duration
+                    .get_or_create(&labels)
+                    .observe(time_taken.as_millis().clamp(0, 10000) as f64 / 1e3);
```

Comment:
Why not use `Duration::as_sec_f64`?

https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_secs_f64

Reply:
Mostly because I didn't know it existed :smile: 

This change would leave it with more precision, which pre-bucketing I suppose might be better.